gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/activity/impl/CommitActivityImpl.java

```diff
@@ -141,22 +151,18 @@ public Callable<Void> apply(final Map.Entry<String, JobState.DatasetState> entry
 
       IteratorExecutor.logFailures(result, null, 10);
 
-      Set<String> failedDatasetUrns = new HashSet<>();
-      for (JobState.DatasetState datasetState : datasetStatesByUrns.values()) {
-        // Set the overall job state to FAILED if the job failed to process any dataset
-        if (datasetState.getState() == JobState.RunningState.FAILED) {
-          failedDatasetUrns.add(datasetState.getDatasetUrn());
-        }
-      }
+      Set<String> failedDatasetUrns = datasetStatesByUrns.values().stream()
+          .filter(datasetState -> datasetState.getState() == JobState.RunningState.FAILED)
+          .collect(HashSet::new, (set, datasetState) -> set.add(datasetState.getDatasetUrn()), HashSet::addAll);
```

Comment:
can you use
```
.map(JobState.DatasetState::getDatasetUrn)
.collect(Collectors.toCollection(HashSet::new))
```
?

Reply:
does this function filter for datasets that are succeeded?